build: add workflow to generate monthly changelog

Planeshifter opened the pull request:
## Description

> What is the purpose of this pull request?

This pull request:

-   adds support for specifying options for the underlying `git log` command used to retrieve the commits for which to parse commits and generate a changelog for, respectively.
- adds a workflow that generates a monthly changelog and pushes it to the [www-blog-monthly-changelog](https://github.com/stdlib-js/www-blog-monthly-changelog) repository.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves https://github.com/stdlib-js/metr-issue-tracker/issues/6.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

Planeshifter committed:
build: add option to parse-commits utility to set git log flags

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter committed:
build: tweak output when flags are supplied and add workflow

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on .github/workflows/generate_monthly_changelog.yml:
```diff
@@ -0,0 +1,136 @@
+        # Pin action to full length commit SHA
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          # Code coverage repository:
```
This comment needs updating.

Planeshifter replied:
Done.

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/generate/README.md:
```diff
@@ -44,13 +44,22 @@ The function returns an object with the following properties:
-To generate a changelog for an upcoming release, provide a valid release type as the second argument.
+The function accepts the following `options`:
+
+-   **releaseType**: a release type for which to generate the changelog for.
+-   **flags**: `git log` options used to retrieve commits to generate the changelog for.
```
```suggestion
-   **releaseType**: a release type for which to generate the changelog.
-   **flags**: `git log` options used to retrieve commits from which to generate the changelog.
```

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/generate/README.md:
```diff
@@ -66,12 +75,32 @@ The following release types are supported:
 -   `auto`: automatically determine the release type based on parsed commit messages.
 -   `none`: no release (equivalent to not specifying a release type).
 
+Under the hood, the changelog leverages `git log` to retrieve the commits from which to assemble the changelog. The `flags` option allows passing any options to the `git log` command, e.g. to generate a changelog for a specified time interval or for an individual contributor.
```
```suggestion
When generating a changelog, the function uses `git log` to retrieve the commits from which to assemble a set of changes. The `flags` option allows passing options to directly to the `git log` command (e.g., to generate a changelog for a specified time interval or for an individual contributor).
```

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/generate/README.md:
````diff
@@ -66,12 +75,32 @@ The following release types are supported:
 
+Under the hood, the changelog leverages `git log` to retrieve the commits from which to assemble the changelog. The `flags` option allows passing any options to the `git log` command, e.g. to generate a changelog for a specified time interval or for an individual contributor.
+
+```js
````
```suggestion
```javascript
```

In stdlib, convention is to use `javascript`, not `js`.

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/generate/README.md:
```diff
@@ -66,12 +75,32 @@ The following release types are supported:
 
 <section class="notes">
 
+-   When `flags` are supplied, the generated changelog skips printing empty releases and uses a flat output format.
```
```suggestion
-   When `flags` are supplied, the generated changelog does not include empty releases and uses a flat output format.
```

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/generate/lib/main.js:
```diff
@@ -315,7 +341,7 @@ function generate( pkg, releaseType ) {
 	}
 
-	if ( isNamespacePkg ) {
+	if ( isNamespacePkg && !opts.flags ) {
```
Why is the `!opts.flags` check necessary here and below?

kgryte replied:
I suggest adding a `format` option rather than an implicit `opts.flags` check used as a proxy.

Planeshifter replied:
Refactored.

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/generate/lib/validate.js:
```diff
@@ -0,0 +1,74 @@
+	if ( hasOwnProp( options, 'flags' ) ) {
+		opts.flags = options.flags;
+		if ( !isPlainObject( opts.flags ) ) {
+			return new TypeError( format( 'invalid option. `%s` option must be a plain object. Option: `%s`.', 'flags', opts.flags ) );
```
```suggestion
			return new TypeError( format( 'invalid option. `%s` option must be an object. Option: `%s`.', 'flags', opts.flags ) );
```

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/generate/test/test.validate.js:
```diff
@@ -0,0 +1,137 @@
+	t.end();
+});
+
+tape( 'the function returns an error if provided a `releaseType` option which is not a string primitive', function test( t ) {
```
```suggestion
tape( 'the function returns an error if provided a `releaseType` option which is not a string', function test( t ) {
```

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/generate/test/test.validate.js:
```diff
@@ -0,0 +1,137 @@
+	t.end();
+});
+
+tape( 'the function returns an error if provided a `flags` option which is not a plain object', function test( t ) {
```
```suggestion
tape( 'the function returns an error if provided a `flags` option which is not an object', function test( t ) {
```

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/parse-commits/README.md:
```diff
@@ -44,6 +44,7 @@ The function accepts the following `options`:
 -   **dir**: directory for which to parse commits. May be either an absolute path or a path relative to the current working directory. Default: current working directory.
 -   **issueURL**: issue URL. Default: `https://github.com/stdlib-js/stdlib/issues/`.
 -   **prURL**: pull request URL. Default: `https://github.com/stdlib-js/stdlib/pull/`.
+-   **flags**: Options passed to invoked `git log` command to retrieve commit messages.
```
```suggestion
-   **flags**: options passed to invoked `git log` command to retrieve commit messages.
```

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/generate/test/test.js:
```diff
@@ -145,13 +153,26 @@ tape( 'the function generates a changelog for a new release', function test( t )
+
+tape( 'the function generates a changelog for a specific author', function test( t ) {
+	var out = generate( '@stdlib/ndarray/base', {
+		'flags': '--author="Athan Reines"'
```
This looks like the wrong format.

Planeshifter replied:
Updated.

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/parse-commits/lib/commits.js:
```diff
@@ -76,13 +76,15 @@ function extractYAMLBlock( str ) {
 *
 * @private
 * @param {string} dir - directory for which to extract commit details
+* @param {string} flags - `git log` options
```
This should be an object.

Planeshifter replied:
Fixed.

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/parse-commits/lib/commits.js:
```diff
@@ -110,13 +112,20 @@ function extractCommits( dir ) {
 	commits = [];
 	args = [ resolve( dir ) ];
+	if ( flags ) {
+		flagNames = Object.keys( flags );
```
Use `utils/keys`.

Planeshifter replied:
Done/

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/parse-commits/lib/main.js:
```diff
@@ -45,12 +45,14 @@ var debug = logger( 'changelog:parse-commits' );
 * @param {string} [options.dir] - root directory
 * @param {string} [options.issueURL] - issue URL
 * @param {string} [options.prURL] - PR URL
+* @param {string} [options.flags] - `git log` options
```
Should be an object.

Planeshifter replied:
Fixed.

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/parse-commits/lib/validate.js:
```diff
@@ -61,6 +63,12 @@ function validate( opts, options ) {
+	if ( hasOwnProp( options, 'flags' ) ) {
+		opts.flags = options.flags;
+		if ( !isPlainObject( opts.flags ) ) {
+			return new TypeError( format( 'invalid option. `%s` option must be a plain object. Option: `%s`.', 'flags', opts.flags ) );
```
```suggestion
			return new TypeError( format( 'invalid option. `%s` option must be an object. Option: `%s`.', 'flags', opts.flags ) );
```

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/parse-commits/test/test.validate.js:
```diff
@@ -138,6 +138,31 @@ tape( 'if provided a `prURL` option which is not a `string`, the function return
 	t.end();
 });
 
+tape( 'the function returns an error if provided a `flags` option which is not a plain object', function test( t ) {
```
```suggestion
tape( 'the function returns an error if provided a `flags` option which is not an object', function test( t ) {
```

Planeshifter committed:
chore: apply suggestions from code review

Co-authored-by: Athan <kgryte@gmail.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
chore: address PR review feedback

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

Planeshifter committed:
build: add format option to changelog generation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: failed
---

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/generate/lib/validate.js:
```diff
@@ -0,0 +1,87 @@
+var hasOwnProp = require( '@stdlib/assert/has-own-property' );
+var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
+var isPlainObject = require( '@stdlib/assert/is-plain-object' );
+var indexOf = require( '@stdlib/utils/index-of' );
```
Rather than `indexOf`, use `array/base/assert/contains`.factory. Then at L34, you can do

```
var isFormat = contains( FORMAT_OPTIONS );
```

and update the check below accordingly.

Planeshifter replied:
Done.

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/generate/lib/validate.js:
```diff
@@ -0,0 +1,87 @@
+* @param {Object} opts - destination for function options
+* @param {Options} options - function options
+* @param {string} [options.releaseType] - release type
+* @param {string} [options.format] - output format (`grouped` or `aggregated`)
```
Prefer not hard coding the list of output formats here. That just increases the possibility of documentation drift. In general, `validate.js` documentation is typically ignorant of concrete option values, as this is not where users would go to understand what is possible. For that, they'd use the docs in `main.js`.

Planeshifter replied:
Addressed.

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/generate/README.md:
```diff
@@ -66,12 +76,50 @@ The following release types are supported:
 
 <section class="notes">
 
+-   When `flags` are supplied, the generated changelog does not include empty releases and uses a flat output format.
```
Is this note still accurate? It is not clear to me from the diff of `main.js` that this is true or that this constraint is necessary. I am mainly referring to the "flat" output format.

kgryte replied:
Also, it is not clear what the "flat" output format actually refers to.

Planeshifter replied:
Good catch! Removed the note.

kgryte commented on .github/workflows/generate_monthly_changelog.yml:
```diff
@@ -0,0 +1,137 @@
+          var generate = require( '@stdlib/_tools/changelog/generate' );
+          var changelog = generate( '@stdlib', {
+            'flags': {
+              'since': '$UNTIL - 1 week',
```
Maybe I am missing something, but is this range actually for the last month or the last `1` week?

Planeshifter replied:
Fixed; had changed it for testing purposes.

kgryte commented on .github/workflows/generate_monthly_changelog.yml:
```diff
@@ -0,0 +1,137 @@
+            'format': 'aggregated'
+          });
+          console.log( changelog.content );
+          " > ./www-blog-monthly-changelog/monthly_changelog_$UNTIL.md
```
The file name should consistently use underscores. In which case, ideally, we'd convert `2025-05-01` to `2025_05_01`.

Planeshifter replied:
Changed accordingly.

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/generate/README.md:
```diff
@@ -44,13 +44,23 @@ The function returns an object with the following properties:
+The function accepts the following `options`:
+
+-   **releaseType**: a release type for which to generate the changelog.
+-   **format**: whether to generate a `grouped` changelog where commits are grouped by package or `aggregated` one.
```
```suggestion
-   **format**: changelog format. Must be one of the following:

    -   `'grouped'`: group commits by package.
    -   `'aggregated'`: TODO:add description.
```

Given the sub-list, you'll need to insert blank lines to satisfy Markdown linting.

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/parse-commits/README.md:
```diff
@@ -44,6 +44,7 @@ The function accepts the following `options`:
 -   **dir**: directory for which to parse commits. May be either an absolute path or a path relative to the current working directory. Default: current working directory.
 -   **issueURL**: issue URL. Default: `https://github.com/stdlib-js/stdlib/issues/`.
 -   **prURL**: pull request URL. Default: `https://github.com/stdlib-js/stdlib/pull/`.
+-   **flags**: options passed to invoked `git log` command to retrieve commit messages.
```
```suggestion
-   **flags**: options passed to `git log` when retrieving commit messages.
```

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/parse-commits/lib/commits.js:
```diff
@@ -110,13 +113,20 @@ function extractCommits( dir ) {
+	if ( flags ) {
+		flagNames = objectKeys( flags );
+		for ( i = 0; i < flagNames.length; i++ ) {
+			args.push( '--'+flagNames[i]+'='+flags[ flagNames[ i ] ] );
```
```suggestion
			args.push( '--'+flagNames[ i ]+'='+flags[ flagNames[ i ] ] );
```

We should use consistent spacing here.

Planeshifter committed:
chore: apply suggestions from code review

Co-authored-by: Athan <kgryte@gmail.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
chore: address PR review feedback

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter committed:
chore: remove lint directive

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/generate/README.md:
```diff
@@ -44,13 +44,28 @@ The function returns an object with the following properties:
+-   **format**: changelog format. Must be one of the following:
+
+    -   `'grouped'`: group commits by package.
+    -   `'aggregated'`: TODO:add description.
```
@Planeshifter Is this TODO intentional?

Planeshifter replied:
Inserted.

kgryte reviewed (commented):
Apart from the one stray TODO, this is looking good.

kgryte commented:
Also, is there a way to avoid the examples CI failures?

Planeshifter committed:
docs: add description for aggregated format

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Planeshifter commented:
@kgryte We don't have a way to omit example code blocks from being run, do we? I believe the problem is that we only checkout a limited number of commits in the workflow and the examples do not work without the commit history. Locally they run without problems for me.

kgryte commented:
Yes, you use the run-disable directive.

Planeshifter commented:
@kgryte Great, wasn't sure whether it was also honored by the make recipes for running examples. Will add.

Planeshifter committed:
chore: add directives to disable running examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/_tools/changelog/generate/README.md:
```diff
@@ -44,13 +46,30 @@ The function returns an object with the following properties:
+-   **format**: changelog format. Must be one of the following:
+
+    -   `'grouped'`: group commits by package.
+    -   `'aggregated'`: display all commits in a flat list without breaking changes up by package.
```
```suggestion
    -   `'aggregated'`: display all commits in a flat list without grouping commits by package.
```

kgryte committed:
docs: update copy

Signed-off-by: Athan <kgryte@gmail.com>
